src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -38,8 +43,11 @@
      * @param users collection of users
      * @return map {user email : user}
      */
-    public Map<String, User> groupByEmail(final List<User> users) {
-        return null;
+    public Map<String, User> groupByEmail(final List<User> users)
+    {
```

Comment:
Let's stick to java code conventions and always put opening curly brace on the same line with method name.

Reply:
public Map<String, User> groupByEmail(final List<User> users) {
        return users.stream()
                .collect(Collectors.toMap(User::getEmail, Function.identity()));

    }